spec/models/article_spec.rb

```diff
@@ -52,6 +58,32 @@
     it { expect(described_class.count - described_class.publicly_visible.count).to be 1 }
   end
 
+  describe '#pinned' do
+    context 'when other already pinned' do
+      subject(:article) { FactoryBot.create(:article, pinned: false) }
```

Comment:
you can also do this `subject(:article) { FactoryBot.build(:article, pinned: true) }`. Then it is not inserted into the database, but you can validate it. If you then just insert another one in the database (like you did below) it will become invalid